modules/RTC/TraceablePeerConnection.js

```diff
@@ -227,15 +317,140 @@ const dumpSDP = function(description) {
     return `type: ${description.type}\r\n${description.sdp}`;
 };
 
+
+/**
+ * Forwards the {@link peerconnection.iceConnectionState} state except that it
+ * will convert "completed" into "connected" where both mean that the ICE has
+ * succeeded and is up and running. We never see "completed" state for
+ * the JVB connection, but it started appearing for the P2P one. This method
+ * allows to adapt old logic to this new situation.
+ * @return {string}
+ */
+TraceablePeerConnection.prototype.getConnectionState = function() {
+    const state = this.peerconnection.iceConnectionState;
+
+    if (state === 'completed') {
+        return 'connected';
+    }
+
+    return state;
+};
+
+/**
+ * Tells whether or not this TPC instance is using Simulcast.
+ * @return {boolean} <tt>true</tt> if simulcast is enabled and active or
+ * <tt>false</tt> if it's turned off.
+ */
+TraceablePeerConnection.prototype.isSimulcastOn = function() {
+    return !this.options.disableSimulcast
+        && this.simulcast.isSupported()
+        && !this.isP2P;
+};
+
+/**
+ * Handles {@link SignalingEvents.PEER_VIDEO_TYPE_CHANGED}
+ * @param {string} endpointId the video owner's ID (MUC nickname)
+ * @param {VideoType} videoType the new value
+ * @private
+ */
+TraceablePeerConnection.prototype._peerVideoTypeChanged
+= function(endpointId, videoType) {
+    // Check if endpointId has a value to avoid action on random track
+    if (!endpointId) {
+        logger.error(`No endpointID on peerVideoTypeChanged ${this}`);
+
+        return;
+    }
+    const videoTrack = this.getRemoteTracks(endpointId, MediaType.VIDEO);
+
+    if (videoTrack.length) {
+        // NOTE 1 track per media type is assumed
+        videoTrack[0]._setVideoType(videoType);
+    }
+};
+
+/**
+ * Handles remote track mute / unmute events.
+ * @param {string} endpointId the track owner's identifier (MUC nickname)
+ * @param {MediaType} mediaType "audio" or "video"
+ * @param {boolean} isMuted the new mute state
+ * @private
+ */
+TraceablePeerConnection.prototype._peerMutedChanged
+= function(endpointId, mediaType, isMuted) {
+    // Check if endpointId is a value to avoid doing action on all remote tracks
+    if (!endpointId) {
+        logger.error('On peerMuteChanged - no endpoint ID');
+
+        return;
+    }
+    const track = this.getRemoteTracks(endpointId, mediaType);
+
+    if (track.length) {
+        // NOTE 1 track per media type is assumed
+        track[0].setMute(isMuted);
+    }
+};
+
+TraceablePeerConnection.prototype.getLocalTracks = function(mediaType) {
+    let tracks = Array.from(this.localTracks.values());
+
+    if (mediaType !== undefined) {
+        tracks = tracks.filter(track => track.getType() === mediaType);
+    }
+
+    return tracks;
+};
+
+/**
+ * Obtains all remote tracks currently known to this PeerConnection instance.
+ * @param {string} [endpointId] the track owner's identifier (MUC nickname)
```

Comment:
Sounds reasonable, but given that there are only 2 arguments and hardly ever one of them is undefined I would rather keep it as is. It will be only extra effort when calling this method to declare each argument inside of an object.